theme.json

```diff
@@ -0,0 +1,413 @@
+        }
+      },
+      "h3": {
+        "typography": {
+          "fontSize": "var(--wp--custom--heading--3--font-size)",
+          "lineHeight": "var(--wp--custom--heading--3--line-height)",
+          "fontWeight": "var(--wp--custom--heading--3--font-weight)",
+          "fontFamily": "var(--wp--custom--heading--3--font-family)",
+          "letterSpacing": "var(--wp--custom--heading--3--letter-spacing)"
+        }
+      },
+      "h4": {
+        "typography": {
+          "fontSize": "var(--wp--custom--heading--2--font-size)",
+          "lineHeight": "var(--wp--custom--heading--2--line-height)",
+          "fontWeight": "var(--wp--custom--heading--2--font-weight)",
+          "fontFamily": "var(--wp--custom--heading--2--font-family)",
+          "letterSpacing": "var(--wp--custom--heading--2--letter-spacing)"
+        }
+      },
+      "h5": {
+        "typography": {
+          "fontSize": "var(--wp--custom--heading--1--font-size)",
+          "lineHeight": "var(--wp--custom--heading--1--line-height)",
+          "fontWeight": "var(--wp--custom--heading--1--font-weight)",
+          "fontFamily": "var(--wp--custom--heading--1--font-family)",
+          "letterSpacing": "var(--wp--custom--heading--1--letter-spacing)"
+        }
+      },
+      "h6": {
+        "typography": {
+          "fontSize": "var(--wp--custom--heading--1--font-size)",
+          "lineHeight": "var(--wp--custom--heading--1--line-height)",
+          "fontWeight": "var(--wp--custom--heading--1--font-weight)",
+          "fontFamily": "var(--wp--custom--heading--1--font-family)",
+          "letterSpacing": "var(--wp--custom--heading--1--letter-spacing)"
+        }
+      },
+      "button": {
+        "color": {
+          "background": "var(--wp--custom--button--primary--default)",
+          "text": "var(--wp--custom--button--primary--label)"
+        },
+        "typography": {
+          "fontSize": "clamp(1.00rem, calc(1.16rem + -0.15vw), 1.13rem)",
+          "lineHeight": "var(--wp--custom--body--2--line-height)",
+          "fontWeight": "var(--wp--custom--body--2--font-weight)",
+          "fontFamily": "var(--wp--custom--body--2--font-family)"
+        },
+        "spacing": {
+          "padding": {
+            "top": "clamp(0.50rem, calc(0.81rem + -0.30vw), 0.75rem)",
+            "bottom": "clamp(0.50rem, calc(0.81rem + -0.30vw), 0.75rem)",
+            "left": "1.50rem",
+            "right": "1.50rem"
+          }
+        },
+        "border": {
+          "radius": "999999px"
+        },
+        ":hover": {
+          "color": {
+            "background": "var(--wp--custom--button--primary--hover)"
+          }
+        },
+        ":active": {
+          "color": {
+            "background": "var(--wp--custom--button--primary--active)"
+          }
+        }
+      }
+    },
+    "blocks": {
+      "core/heading": {
+        "color": {
+          "text": "var(--wp--custom--text--primary)"
+        },
+        "elements": {
```

Comment:
I don't think these are necessary since the element styles are getting declared above. Does it get messed up if you don't do it here? I can't remember why these are here. It might be so that the styles update to a new default if you select a new heading level in the block ... but I think that should probably fall back to the global defaults. I'd try removing these and seeing what happens if you haven't already.

Reply:
Lemme check this once more as I did try to simplify this down a lot.

Reply:
Confirmed, works just fine without it. I thought I had seen some difference so I kept it, but guess not. This saves a bunch of room in the json file!